src/docs/binding-generics.md

```diff
@@ -437,3 +437,191 @@ We could also consider solving this on the runtime level.
 ### Generic classes
 
 TBD
+
+## Protocol constraints
+
+This section describes how to project protocol constraints on generic type parameters from Swift to C\#. The goal here is not to describe full projections of protocols; for that, please refer to the [binding protocols doc](binding-protcols.md).
+
+A Swift protocol can be mapped to C\# as an interface. Then Swift generic parameters with protocol constraint can be mapped into C\# as generic parameters with interface constraint. This should work for both functions and types.
+
+### Protocol constraints on functions
+
+Consider the following simple example in Swift:
+
+```swift
+public protocol Printable {
+ func printMe()
+}
+
+public func printPrintable<T: Printable>(data: T) {
+ data.printMe()
+}
+
+// Equivalent to: public func printPrintable<T>(data: T) where T : Printable
+```
+
+This can be projected into C\# as follows:
+
+```csharp
+public interface ISwiftPrintableProtocol
```

Comment:
Naming nit: if the protocol is named `Printable` in Swift, we can named it `IPrintable` in C#. We'd like the resulting binding to line up as closely to the original as possible while still feeling like a real C# type.